### ISSUE TITLE
[steamshovel] loading matplotlib artists from saved session segfaults (Trac #2048)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2040]
With Qt 5.9.0 on OS X Sierra, steamshovel segfaults on startup if there were any matplotlib artists in the saved session state, as rendering these attempts to call QGLWidget::bindTexture() before the GL context is fully initialized. Delaying session loading until the main window is shown (as in the attached patch) works around this behavior. I'm not sure if this is a Qt bug or OS X bug.

Steps to reproduce:
1. Start a new steamshovel session with `steamshovel --reset GCD.i3 data.i3`
2. Add Geometry and Waveform artists. Bind the Waveform artist to !InIceRawData.
3. Quit steamshovel and start again with `steamshovel GCD.i3 data.i3`. It will segfault before the main window is shown. 

Migrated from https://code.icecube.wisc.edu/ticket/2048
```json
{
    "status": "closed", 
    "changetime": "2017-07-03T09:30:50", 
    "description": "With Qt 5.9.0 on OS X Sierra, steamshovel segfaults on startup if there were any matplotlib artists in the saved session state, as rendering these attempts to call QGLWidget::bindTexture() before the GL context is fully initialized. Delaying session loading until the main window is shown (as in the attached patch) works around this behavior. I'm not sure if this is a Qt bug or OS X bug.\n\nSteps to reproduce:\n1. Start a new steamshovel session with `steamshovel --reset GCD.i3 data.i3`\n2. Add Geometry and Waveform artists. Bind the Waveform artist to !InIceRawData.\n3. Quit steamshovel and start again with `steamshovel GCD.i3 data.i3`. It will segfault before the main window is shown. ", 
    "reporter": "jvansanten", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1499074250139553", 
    "component": "combo core", 
    "summary": "[steamshovel] loading matplotlib artists from saved session segfaults", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2017-06-30T13:23:13", 
    "milestone": "", 
    "owner": "sander.vanheule", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170630T13:23:38: 
* jvansanten changed attachment from "" to "delay_session.diff"
* jvansanten commented:

> Patch to delay session loading

**Comment by icecube-trac** [comment]
Trac update at 20170630T19:18:30: 
* olivas changed owner from "" to "sander.vanheule"
* olivas changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20170703T09:30:50: 
* sander.vanheule commented:

> In changeset 156903:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="156903"
> Delay session loading. Fixes #2040.
> 
> Prevent segfault by delaying the restoring of all Artists until the GL context
> has been properly initialised.
> 
> Patch provided by Jakob Van Santen <jvansanten@icecube.wisc.edu>
> ```


* sander.vanheule changed status from "assigned" to "closed"
* sander.vanheule changed resolution from "" to "fixed"
